Objective-C/CBLDatabase.mm

```diff
@@ -1041,11 +1041,22 @@ - (void) scheduleDocumentExpiration: (NSTimeInterval)minimumDelay {
 
 - (void) purgeExpiredDocuments {
     dispatch_async(dispatch_get_global_queue( DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        CBLLog(Database, @"Purging expired documents...");
-        C4Error err;
-        UInt64 nPurged = c4db_purgeExpiredDocs(_c4db, &err);
-        CBLLog(Database, @"Purged %lld expired documents", nPurged);
-        [self scheduleDocumentExpiration: 1.0];
+        CBL_LOCK(self) {
+            if (!_c4db)
+                return;
+            
+            CBLLog(Database, @"Purging expired documents...");
+            C4Error c4err;
+            UInt64 nPurged = c4db_purgeExpiredDocs(_c4db, &c4err);
+            if (nPurged >= 0)
+                CBLLog(Database, @"Purged %lld expired documents", nPurged);
+            else {
+                NSError* error;
+                convertError(c4err, &error);
+                CBLWarnError(Database, @"Cannot purge expired documents: %@", error);
```

Comment:
if we warn about purge without any docs, then we might need to re-write the schedule with a 1 second delay. Or else the console will print warning when the next expiry is less than a second(and the delay is 0.0, which is basically they set an expiry Date().now).

https://github.com/couchbase/couchbase-lite-ios/blob/master/Objective-C/CBLDatabase.mm#L1029

Reply:
We warn only when the nPurge < 0 which means there is an error.

Reply:
yea thats right.